Audio added for new/lost subscribers & new video

ejbarrow opened the pull request:
I added audio to the project. By default on load the audio is switched off, but on the click of the audio button next to channel name the audio will toggle on/off.

When audio is toggled on, a bell will sound when a new subscriber is detected, and another sound for when the project detects a lost subscriber. I have also added a sound for when the project detects the YouTuber has uploaded a new video.

Index.html has been altered to add an audio image for toggling. as well as adding audio attributions. Script.js has been altered to keep tabs on the subscriber and video count and play a sound when a change is noticed.

This change is useful especially for new YouTubers that maybe cannot leave the project open on a separate screen, but can receive an audio cue to check the project in their browser tab for new subscribers.

An example of this change is shown here (for PewDiePie) : http://hypertora.com/livesubcountertest/#!/UC-lHJZR3Gqxm24_Vd_AJ5Yw

akshatmittal committed:
Initial commit

akshatmittal committed:
First Commit

akshatmittal committed:
Fix readme.md

akshatmittal committed:
Final v1

akshatmittal committed:
Sorry Marques

akshatmittal committed:
Analytics + LazyLoad

akshatmittal committed:
Analytics + LazyLoad

akshatmittal committed:
Spelling, oops

akshatmittal committed:
Few fixes

akshatmittal committed:
Few fixes

akshatmittal committed:
Update README.md

akshatmittal committed:
Better load time - move Analytics

akshatmittal committed:
Merge branch 'gh-pages' of https://github.com/akshatmittal/youtube-realtime into gh-pages

akshatmittal committed:
Remove Tweet count, since it's always incorrect because Twitter

akshatmittal committed:
Alternate Access

akshatmittal committed:
Remove Viewport, better on mobile

akshatmittal committed:
Different YouTube Regions

akshatmittal committed:
Fix prev commit

akshatmittal committed:
Google Analytics now tracks usernames

akshatmittal committed:
Fix Secure Fonts

akshatmittal committed:
Fixes

akshatmittal committed:
Merges #3, plus minor fixes

akshatmittal committed:
Added support for Channels, minor fixes

akshatmittal committed:
Apparently Firefox is the only major browser to not support innerText anymore

akshatmittal committed:
YouTube Realtime v2.0

akshatmittal committed:
YouTube Realtime v2.0

akshatmittal committed:
YouTube Realtime v2.0

akshatmittal committed:
Immersive Mode (beta)

akshatmittal committed:
Improved Immersive Mode and Overall responsiveness of the website

akshatmittal committed:
Update icon (high-res)

akshatmittal committed:
Bots just love to cache content, but you gotta bust them.

akshatmittal committed:
New API Keys to handle traffic

akshatmittal committed:
New Twitter

akshatmittal committed:
Big Update

akshatmittal committed:
Fix for breaking changes in YouTube API

Commit:
Embeds

akshatmittal committed:
New Design

akshatmittal committed:
Embeds

akshatmittal committed:
New Design

akshatmittal committed:
Fix Async loads

akshatmittal committed:
Bug Fixes

akshatmittal committed:
Dark Theme

akshatmittal committed:
Fix Embeds Bug

ejbarrow committed:
Audio added for new/lost subscribers & new video

ejbarrow committed:
Audio promts added - url fix

ejbarrow committed:
Spelling mistake on index fix

akshatmittal commented:
I believe most people won't prefer audio for this project. Also, the enable/disable prompt should preferably be at the bottom. 

I was actually planning to implement a counter for people who just stick it there, where the project would show the change, up/down in the title bar itself. That I believe would be the preferred way.

ejbarrow committed:
Audio toggle moved to lower part of page, added notifcation to page title ability, adited baseURL to automatically take url from window

ejbarrow commented:
1.Audio toggle moved to the bottom of the page.  This is disabled by default. I agree most may not use it, but useful for small YouTubers that may have it open in a different window or (like me) have so many tabs open you can't read the tab title. Toggle updated to save toggle status in a cookie.

2.Added notifications in the title bar. Can be toggled on/off. When the page does not have focus, then the title is edited to show subscribers gained/lost since the page last had focus. This also saves toggle status in a cookie. Switched off by default. Toggle located in the lower area of the page with other toggles. e.g "(+3) ...'s Realtime Subscriber Count on YouTube"

3. Base URL code edited to automatically take page URL from the window. This is better for users hosting their own version of the project, and means only the API keys need updating. If URL is not updated then it automatically redirects to URL in the JS.

Changes are reflected in demo at http://hypertora.com/livesubcountertest/#!/UC-lHJZR3Gqxm24_Vd_AJ5Yw

ejbarrow committed:
Double Minus Fix, Page focus update Fix

ejbarrow committed:
0 sub change tab notification removed - only lost and gained subs show in tab
